examples/kind/kind_with_config_test.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kind
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"path/filepath"
+	"testing"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/homedir"
+	"sigs.k8s.io/e2e-framework/klient/conf"
+	"sigs.k8s.io/e2e-framework/support/kind"
+)
+
+var (
+	kubeconfig   string
+	dep          *appsv1.Deployment
+	clientset    *kubernetes.Clientset
+	count        uint64
+	replicaCount int32 = 2
+	ctx                = context.TODO()
+	cfg          *rest.Config
+	namespace    *corev1.Namespace
+	kc           *kind.Cluster
+)
+
+func TestCreateWithConfig(t *testing.T) {
+	setup()
+	initializeResObjects()
+	teardown()
+}
+
+func setup() {
```

Comment:
For this example, I don't think you need to expose `kubeconfig` as a flag since kind will write its cluster context in the default kubeconfig path (home/.kube/config).